media-video/gaupol/gaupol-1.14.1-r1.ebuild

```
@@ -0,0 +1,81 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit distutils-r1 virtualx xdg-utils
+
+DESCRIPTION="A subtitle editor for text-based subtitles"
+HOMEPAGE="https://otsaloma.io/gaupol/ https://github.com/otsaloma/gaupol/"
+SRC_URI="https://github.com/otsaloma/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE="+player spell"
+
+RDEPEND="
+	app-text/iso-codes
+	dev-python/chardet[${PYTHON_USEDEP}]
+	dev-python/pygobject:3[${PYTHON_USEDEP}]
+	x11-libs/gtk+:3[introspection]
+	player-integrated? (
+		>=media-libs/gstreamer-1.6
+		media-libs/gst-plugins-bad
+		media-libs/gst-plugins-base[pango]
+		media-libs/gst-plugins-good
+		media-plugins/gst-plugins-gtk
+		media-plugins/gst-plugins-libav
```

```suggestion
		>=media-libs/gstreamer-1.6[introspection]
		media-libs/gst-plugins-base[pango]
		media-plugins/gst-plugins-gtk
```

Or move this into optfeature as `optfeature "internal player support" "media-libs/gstreamer[introspection] media-libs/gst-plugins-base[pango] media-plugins/gst-plugins-gtk"`

---

If some codecs and container formats need specific plug-in's then they should be optfeatures. Or the should just be some postinst message saying that you should look into gstreamer plugins for missing features.

---

No, that's not how you depend on GStreamer plugins.

You depend on specific plugins **if and only if** gaupol specifically enforces them, i.e. it can't work without them. If it doesn't (i.e. lets GStreamer use any plugin for given format), then you do not add dependencies and instead let users enable plugins via USE flags on `gst-plugins-meta`.